src/main/ast/Literal.kt

```diff
@@ -0,0 +1,14 @@
+package ast
+
+import exceptions.RegexSyntaxException
+
+class Literal(val characters : String) : Group("literal") {
+    
+    override fun render(builder: StringBuilder) {
+        if (characters.isEmpty()) {
+            throw RegexSyntaxException("Literal should not be empty.")
+        } else {
+            builder.append(characters)
+        }
```

Comment:
Consider using [`require`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/require.html) to make code less verbose.

Reply:
That would switch to IllegalArgumentException instead of something internal - are we ok with that?

Reply:
Yeah, I think the use of `require` is pretty common within Kotlin anyway.